### DISCUSSION TITLE
Reality vs expectations

### DISCUSSION THREAD

**Discussion by dzikiBOB** [tony-fav/tasmota-blerry#42] [Q&A]
Gentlemen, I bought an inkbird sensor to monitor the refrigerator. Before running the berry script, Tasmota 11.0.0 detects 12 BLE devices (8x ATC, 3xFLORA, 1x MJ_HTv1), with a complete parameter list for each sensor. Of course it doesn't detect inkbird, but that's what I expected.
On the other hand, after running the berry script, Tasmota sees the inkbird but only 7 other devices (4x ATC, 3x FLORA), of which the results of the flora devices are severely reduced, showing either moisture or fertility (no temperature, illuminance and battery).
Is this normal behavior? I admit that I did not expect it :)

The second question is about transferring inkbird data. The tasmot screen shows readable parameters: temperature, humidity, battery dewpoint, while in the console everything looks coded:
tele / k / blerry / BLE = {"DetailsBLE": {"mac": "49211216ABB9", "a": "fridge", "RSSI": - 76, "p": "0201060302F0FF04097370730AFFDD02EA0700EA6F5206"}}

As such, sending data via MQTT is probably useless when using Openhab. I know that all of you here are using HomeAssistant, but maybe you will consider sending the decoded reading from the sensor via MQTT?

**Comment by tony-fav** [comment]
Blerry does not auto detect devices. You have to add them manually to the `blerry_config.json` I do not think we have support for the MJ_HT_V1 unless it is flashed to the pvvx firmware. Similarly, we have only recently supported one of the mi flora devices (https://github.com/tony-fav/tasmota-blerry/issues/33).

It sounds as though your Blerry configuration is not complete and is not even running at all perhaps erroring after setting some of the BLE Aliases.

The entire purpose of Blerry is to send the decoded BLE packet over mqtt. The `base_topic` parameter controls where it goes. By default, this is `tele/tasmota_blerry/...` If you have an error in your setup, you will not be getting any messages there. The message you are seeing is the message Tasmota sends simply saying "I heard a BLE advertisement with this raw data." Blerry takes the data in that message and converts and republishes with the parsed data. For example,

On topic `tele/tasmota_blerry/ATC3_Office`
```json
{
  "RSSI": -71,
  "Triggered_by_Humidity": "OFF",
  "Humidity": 43.6,
  "Temperature": 22.79,
  "RSSI_via_plug_kitchen_bugZapper": -71,
  "DewPoint": 9.74,
  "Count": 134,
  "via_device": "plug_kitchen_bugZapper",
  "GPIO_PA5": "OFF",
  "Battery_Voltage": 2.772,
  "mac": "A4C138A8E41B",
  "Time": "2022-04-12T10:42:56",
  "GPIO_PA6": "ON",
  "Triggered_by_Temperature": "ON",
  "Time_via_plug_kitchen_bugZapper": "2022-04-12T10:42:56",
  "Flag": 5,
  "Battery": 63,
  "alias": "ATC3_Office"
}
```

**Comment by dzikiBOB** [comment]
hi Tony,
Thank you for the answer and for explaining the method of operation. I did a reinstallation from scratch and for fridge temperature I had decoded messages posted just like in your ATC3_Office.
But then I wanted to add my other 12 devices so that they were also visible and sent to the MQTT Bridge, therefore I created a configuration file:
`{
  "devices": {
    "4921121606B9": {
      "alias": "fridge",
      "model": "IBSTH2",
      "publish_attributes": true
    },
    "A4C1380B8A58": {
      "alias": "mi_syp",
      "model": "ATCpvvx",
      "publish_attributes": true
    },
    "A4C138675CC6": {
      "alias": "mi_salre",
      "model": "ATCpvvx"
    },
    "A4C13810AEE6": {
      "alias": "mi_dzi",
      "model": "ATCpvvx"
    },
    "A4C138987532": {
      "alias": "mi_wc",
      "model": "ATCpvvx"
    },
    "A4C138887989": {
      "alias": "mi_laz",
      "model": "ATCpvvx"
    },
    "A4C13880058F": {
      "alias": "mi_susz",
      "model": "ATCpvvx"
    },
    "A4C138DA736B": {
      "alias": "mi_kuch",
      "model": "ATCpvvx"
    },
    "A4C1385D574E": {
      "alias": "mi_hol",
      "model": "ATCpvvx"
    },
    "C47C8D6CB35B": {
      "alias": "flo_hib_du",
      "model": "Xiaomi"
    },
    "C47C8D6DEFCC": {
      "alias": "flo_hib_ma",
      "model": "Xiaomi"
    },
    "C47C8D6DEBCC": {
      "alias": "flo_lil",
      "model": "Xiaomi"
    },
    "582D34343DB2": {
      "alias": "h1j_st",
      "model": "Xiaomi"
    }
  },
  "override": {
    "publish_attributes": true,
    "discovery": false,
    "base_topic": "tele/km_blerry",
    "via_pubs": true,
    "precision": {
      "Temperature": 1,
      "Humidity": 0,
      "Battery": 0
    }
  }
}`

Do you think something is wrong with this code? (not sure what for are "publish_attributes"/"via_pubs"/"discovery").
 The point is that when using such a configuration, the sensor from the refrigerator again is not decoded by blerry (although it is visible in the tasmota console
`Time ":" 2022-04-27T13: 09: 44 "," BLEDevices ": {" total ": 19," 5448E68D6B4B ": {" i ": 0," r ": - 72}," 4921121606B9 ": { "i": 1, "r": - 74, "a": "fridge"}...`

Or maybe 13 sensors are too many for BLErry to process it?
